homeassistant/components/connector/__init__.py

```diff
@@ -0,0 +1,109 @@
+"""The Connector integration."""
+from __future__ import annotations
+
+import asyncio
+from datetime import timedelta
+import logging
+from socket import timeout
+
+from connectorlocal.connectorlocal import WIFIMOTORTYPE, ConnectorHub
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_API_KEY, CONF_HOST, EVENT_HOMEASSISTANT_STOP
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import device_registry as dr
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from .const import (
+    DOMAIN,
+    KEY_COORDINATOR,
+    KEY_GATEWAY,
+    KEY_MULTICAST_LISTENER,
+    MANUFACTURER,
+    PLATFORMS,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up connector from a config entry."""
+    hass.data.setdefault(DOMAIN, {})
+    host = entry.data[CONF_HOST]
+    key = entry.data[CONF_API_KEY]
+    connector = ConnectorHub(ip=host, key=key)
+    hub_list = None
+
+    if KEY_MULTICAST_LISTENER not in hass.data[DOMAIN]:
+        connector.start_receive_data()
+        hub_list = await connector.device_list()
+        hass.data[DOMAIN][KEY_MULTICAST_LISTENER] = connector
+
+        def stop_motion_multicast(event):
+            """Stop multicast thread."""
+            _LOGGER.debug("Shutting down Connector Listener")
+            connector.close_receive_data()
+
+        hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STOP, stop_motion_multicast)
+
+    def update_gateway():
+        """Call all updates using one async_add_executor_job."""
+        for device in hub_list.values():
+            try:
+                device.update_blinds()
+            except timeout:
+                _LOGGER.warning("Update gateway timeout")
+
```

Comment:
Either make this private or move it as an inner function of `async_update_data`

Reply:
Thanks for your advice